src/commands/list_sweep.adb

```diff
@@ -170,52 +156,30 @@ procedure List_Sweep (Pa : in out Parse_Array; Pa_Last : in out Integer) is
             end if;
 
             --  Check for V SEMIDEP    and demand PASSIVE ex Perf
-            if De.Part.V.Kind = Semidep then
-               if (Pr.IR.Qual.Verb.Tense_Voice_Mood.Voice = Passive)  and
-                 (Pr.IR.Qual.Verb.Tense_Voice_Mood.Tense in Pres .. Fut)  and
-                 (Pr.IR.Qual.Verb.Tense_Voice_Mood.Mood in Ind .. Imp)
-               then
-                  Allowed := False;
-               elsif (Pr.IR.Qual.Verb.Tense_Voice_Mood.Voice = Active)  and
-                 (Pr.IR.Qual.Verb.Tense_Voice_Mood.Tense in Perf .. Futp)  and
-                 (Pr.IR.Qual.Verb.Tense_Voice_Mood.Mood in Ind .. Imp)
-               then
-                  Allowed := False;
-               else
-                  null;
-               end if;
+            if De.Part.V.Kind = Semidep and
+               (Pr.IR.Qual.Verb.Tense_Voice_Mood.Mood in Ind .. Imp) and
+               (((Pr.IR.Qual.Verb.Tense_Voice_Mood.Voice = Passive) and
+                 (Pr.IR.Qual.Verb.Tense_Voice_Mood.Tense in Pres .. Fut)) or
+                ((Pr.IR.Qual.Verb.Tense_Voice_Mood.Voice = Active) and
+                 (Pr.IR.Qual.Verb.Tense_Voice_Mood.Tense in Perf .. Futp)))
+            then
+               Allowed := False;
             end if;
 
             if  Words_Mdev (For_Word_List_Check)  then
                if (Pr.IR.Qual.Verb.Person = 1) and then
                  (Pr.IR.Qual.Verb.Number = S)
                then
-                  if ((De.Part.V.Kind in X .. Intrans)  and
-                    (Pr.IR.Qual.Verb.Tense_Voice_Mood =
-                    (Pres, Active, Ind))) or else
-                    ((De.Part.V.Kind = Dep)  and
-                    (Pr.IR.Qual.Verb.Tense_Voice_Mood =
-                    (Pres, Passive, Ind))) or else
-                    ((De.Part.V.Kind = Semidep)  and
-                    (Pr.IR.Qual.Verb.Tense_Voice_Mood = (Pres, Active, Ind)))
-                  then
-                     Allowed := True;
-                  elsif (De.Part.V.Kind = Perfdef)  and
-                     (Pr.IR.Qual.Verb.Tense_Voice_Mood = (Perf, Active, Ind))
-                  then
-                     Allowed := True;
-                  else
-                     Allowed := False;
-                  end if;
+                  Allowed := (Pr.IR.Qual.Verb.Tense_Voice_Mood =
+                     (Pres, Active, Ind)) and
+                    ((De.Part.V.Kind in X .. Intrans) or else
+                     (De.Part.V.Kind = Dep) or else
+                     (De.Part.V.Kind = Semidep) or else
+                     (De.Part.V.Kind = Perfdef));
                elsif De.Part.V.Kind = Impers then
-                  if (Pr.IR.Qual.Verb.Person = 3)  and then
+                  Allowed := (Pr.IR.Qual.Verb.Person = 3)  and then
```

Comment:
A typo - should be "Allowed =". Assignments are not permitted inside expressions (compile time error).